src/editors/TextEditor/TextEditor.jsx

```diff
@@ -27,24 +46,27 @@ const TextEditor = () => {
         )
         : (
           <Editor
-            onInit={(evt, editor) => { editorRef.current = editor; }}
+            onInit={(evt, editor) => {
+              editorRef.current = editor;
+            }}
             initialValue={blockValue ? blockValue.data.data : ''}
             init={{
-              height: '100%',
+              setup: (editor) => {
+                editor.ui.registry.addButton('imageuploadbutton', {
+                  icon: 'image',
+                  onAction: () => openUploadModal(),
+                });
+              },
+              plugins: 'link codesample emoticons table charmap code autoresize',
               menubar: false,
-              plugins: [
-                'advlist autolink lists link image charmap print preview anchor',
-                'searchreplace visual blocks code fullscreen',
-                'insertdatetime media table paste code help wordcount',
-                'autoresize',
-              ],
               toolbar: 'undo redo | formatselect | '
             + 'bold italic backcolor | alignleft aligncenter '
-            + 'alignright alignjustify | bullist numlist outdent indent | '
-            + 'removeformat | help',
+            + 'alignright alignjustify | bullist numlist outdent indent |'
+            + 'imageuploadbutton | link | emoticons | table | codesample | charmap |'
+            + 'removeformat | hr |code',
```

Comment:
Hmmm - yet another place the plugins are listed - and this time in a specific order. I see that all these complications make what's implemented much easier than a configured list. But it all seems too hard-coded.